### DISCUSSION TITLE
Add local package management option

### DISCUSSION THREAD

**Discussion by OpenSourceAnarchist** [vlang/v#11685] [Ideas]
Currently, when V installs dependencies, it only allows for packages to be installed globally. This differs from other package managers like `npm` (I think) and `cargo` which default to downloading and building packages in the current project folder. I propose adding a way to specify local dependency/package installation in the `.mod` file of the project and when using `v install` as an optional flag.

This will still have packages install globally as the default behavior (like it is now) but will add similar functionality to other languages' package managers and dependency resolution features.

**Comment by nedpals** [comment]
This was the purpose of the merged PR #2891 so it should be easy to implement an option to install modules locally.

**Comment by medvednikov** [comment]
What should be the flag's name?

```
v install --local ui
```
?

**Comment by JalonSolov** [comment]
Seems reasonable for `v install`

The other half was wanting v to automatically install locally if there was a marker of some kind in the v.mod file, so that just an `import` statement in the code would cause v to install locally instead of in ~/.vmodules when it compiles.
